storage/local/storage.go

```diff
@@ -1291,6 +1322,122 @@ func (s *memorySeriesStorage) calculatePersistenceUrgencyScore() float64 {
 	return score
 }
 
+// quarantineSeries registers the provided fingerprint for quarantining. It
+// always returns immediately. Quarantine requests are processed
+// asynchronously. If there are too many requests queued, they are simply
+// dropped.
+//
+// Quarantining means that the series file is moved to the orphaned directory,
+// and all its traces are removed from indices. Call this method if an
+// unrecoverable error is detected while dealing with a series, and pass in the
+// encountered error. It will be saved as a hint in the orphaned directory.
+func (s *memorySeriesStorage) quarantineSeries(fp model.Fingerprint, metric model.Metric, err error) {
+	req := quarantineRequest{fp: fp, metric: metric, reason: err}
+	select {
+	case s.quarantineRequests <- req:
+		// Request submitted.
+	default:
+		log.
+			With("fingerprint", fp).
+			With("metric", metric).
+			With("reason", err).
+			Warn("Quarantine queue full. Dropped quarantine request.")
+		s.seriesOps.WithLabelValues(droppedQuarantine).Inc()
+	}
+}
+
+func (s *memorySeriesStorage) handleQuarantine() {
+	for {
+		select {
+		case req := <-s.quarantineRequests:
+			s.purgeSeries(req.fp, req.metric, req.reason)
+			log.
+				With("fingerprint", req.fp).
+				With("metric", req.metric).
+				With("reason", req.reason).
+				Warn("Series quarantined.")
+		case <-s.quarantineStopping:
+			log.Info("Series quarantining stopped.")
```

Comment:
This is in parallel to all the other "foobar stopped." messages. We can downgrade them all to debug, but it should happen consistently.

Do you want me to do this mingled with the rest of the changes? In general, I'd prefer a separate change to de-spam logging. (There are more messages to be thought through.)